### PR TITLE
Install things as necessary from shinycoreci r-universe

### PR DESCRIPTION
**Pull request by schloerke**
Install dependencies as necessary when testing apps.  This should revive older R versions when testing!

Very cool to see that old windows runs couldn't test 063 due to install failures!

- [ ] Figure out why old macos is not finishing.
  * I could not figure it out. Am punting as it's older R versions on macOS
- [ ] Followup PR - Fix tests on all apps; Add skip statements
- [x] Make sure ubuntu docker can build!
- [ ] Followup PR - Add support for macOS brew dependencies. (Or just hard code install them!)

### CODE REVIEW

**Comment by schloerke** [comment]
- [ ] FIgure out why old macos is not finishing. (Difficult?)

File: .github/internal/install-shinyverse/action.yaml [hunk 1]
```diff
@@ -56,38 +74,53 @@ runs:
...
+        value <-
+          if (short_r_version == "4.4") {
+            "44"
+          } else if (short_r_version == "4.3") {
+            "43"
+          } else if (short_r_version == "4.2") {
+            "42"
+          } else {
+            ""
+          }
```
Comment (cpsievert) [inline .github/internal/install-shinyverse/action.yaml:93]: Would it make sense to try and future-proof this?

```suggestion
        value <- if (r_version >= "4.2") {
          sub(".", "", short_r_version, fixed = TRUE)
        } else {
          ""
        }
```

Comment (schloerke) [reply .github/internal/install-shinyverse/action.yaml:93]: I believe we can drop it entirely. 

https://github.com/rstudio/shiny-workflows/commit/4a08b567b41417f00a19b0f63897ba89e91282ba 

^^ will be doing automatic support for the rtools version

Comment (schloerke) [reply .github/internal/install-shinyverse/action.yaml]: Dropped. **resolving

File: README.md [hunk 2]
```diff
@@ -30,45 +30,47 @@ shinycoreci [uses the following GitHub Runnner Images](https://github.com/rstudi
...
+- [rstudio/shinytest2](https://github.com/rstudio/shinytest2)
+- [rstudio/shinythemes](https://github.com/rstudio/shinythemes)
+- [rstudio/shinyvalidate](https://github.com/rstudio/shinyvalidate)
+- [rstudio/thematic](https://github.com/rstudio/thematic)
+- [rstudio/webdriver](https://github.com/rstudio/webdriver)
+- [rstudio/websocket](https://github.com/rstudio/websocket)
+
+#### R-Universe
+
+`{shinycoreci}` testing leverages rOpenScie [`r-universe`](https://r-universe.dev/search/), specifically the [`posit-dev-shinycoreci`](https://posit-dev-shinycoreci.r-universe.dev/builds) universe. This universe is used to install the latest development versions of the Shiny related packages (updated hourly) used in the testing apps without the need for a GitHub token. This last detail is important, as it allows GitHub Actions to install packages freely without the worry of being rate limited. This gives us the ability to attempt to install each app’s dependencies independently, leading to higher test coverage as a single dependencies does not block the entire test execution.
```
Comment (cpsievert) [inline README.md:73]:
```suggestion
`{shinycoreci}` testing leverages rOpenSci [`r-universe`](https://r-universe.dev/search/), specifically the [`posit-dev-shinycoreci`](https://posit-dev-shinycoreci.r-universe.dev/builds) universe. This universe is used to install the latest development versions of the Shiny related packages (updated hourly) used in the testing apps without the need for a GitHub token. This last detail is important, as it allows GitHub Actions to install packages freely without the worry of being rate limited. This gives us the ability to attempt to install each app’s dependencies independently, leading to higher test coverage as a single dependencies does not block the entire test execution.
```

File: inst/apps/000-pkg-versions/app.R [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      remoteType = get_field("RemoteType"),
+      remotePkgRef = get_field("RemotePkgRef"),
+      remoteRepos = get_field("RemoteRepos"),
+      remotePkgPlatform = get_field("RemotePkgPlatform"),
+    )
+  }
+
+  dt <- bind_rows(lapply(pkgs, get_pkg_info))
+
+  print(dt, n = Inf, width = 1000)
```
Comment (cpsievert) [inline inst/apps/000-pkg-versions/app.R:76]:
```suggestion
```

File: inst/apps/000-pkg-versions/app.R [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  output$all_pkg_info <- renderPrint({
+    # pkg_info <- sessioninfo::package_info("installed", include_base = FALSE)
+
+    # pkg_info %>%
+    #   dplyr::filter(package %in% pkgs, ) %>%
+    #   dplyr::filter
+    #   dplyr::select(package, version, loaded, attached, libPath)
+
+    # sub_pkg_info <- pkg_info[pkg_info$package %in% pkgs, ]
+
```
Comment (cpsievert) [inline inst/apps/000-pkg-versions/app.R:117]:
```suggestion
```

**Review by cpsievert — APPROVED** [review]
Thank you!

**Comment by schloerke** [comment]
@cpsievert any idea on why the older mac runs are failing to install?

It's failing at install dev deps and make lock file within pak

File: .github/internal/install-shinyverse/action.yaml [hunk 5]
```diff
@@ -20,7 +20,19 @@ inputs:
...
+    default: "3.x"
+    required: false
+  packagemanager-cran-url:
+    description: packagemanager url to be used for CRAN installations
+    default: "https://packagemanager.posit.co/cran/latest"
+    required: false
+  shinyverse-cran-url:
+    description: CRAN url to be used for shinyverse installations via r-universe
+    default: "https://posit-dev-shinycoreci.r-universe.dev"
+    required: false
```
Comment (cpsievert) [inline .github/internal/install-shinyverse/action.yaml:32]:
```suggestion
```

File: .github/workflows/apps-test-os.yml [hunk 6]
```diff
@@ -126,39 +131,19 @@ jobs:
...
+        timeout-minutes: 30 # Should be < 10 mins
         uses: ./.github/internal/install-shinyverse
         with:
           r-version: ${{ inputs.r-version }}
           cache-version: ${{ inputs.cache-version }}
           http-user-agent: ${{ inputs.http-user-agent }}
           pandoc-version: ${{ inputs.pandoc-version }}
           extra-packages: ${{ inputs.extra-packages }}
+          packagemanager-cran-url: ${{ inputs.packagemanager-cran-url }}
+          shinyverse-cran-url: ${{ inputs.shinyverse-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-test-os.yml:144]:
```suggestion
```

File: .github/workflows/apps-test-os.yml [hunk 7]
```diff
@@ -27,17 +27,22 @@ on:
...
+      packagemanager-cran-url:
+        type: string
+        description: packagemanager url to be used for CRAN installations
+        default: "https://packagemanager.posit.co/cran/latest"
+        required: false
+      shinyverse-cran-url:
+        type: string
+        description: CRAN url to be used for shinyverse installations via r-universe
+        default: "https://posit-dev-shinycoreci.r-universe.dev"
         required: false
```
Comment (cpsievert) [inline .github/workflows/apps-test-os.yml:41]:
```suggestion
```

File: .github/internal/install-shinyverse/action.yaml [hunk 8]
```diff
@@ -40,54 +48,43 @@ runs:
...
+    #   run: |
+    #     source("R/data-apps-deps.R")
+    #     cat("pkgs=", apps_deps, "\n", file = Sys.getenv("GITHUB_OUTPUT"), sep = "", append = TRUE)
+    # # Must be done as packages are installed freely during testing
+    # - name: Install macOS brew dependencies for all apps
+    #   uses: rstudio/shiny-workflows/setup-macOS-dependencies@v1
+    #   if: ${{ runner.os == 'macOS' }}
+    #   with:
+    #     extra-packages: |
+    #       ${{ steps.app-deps.outputs.pkgs }}
```
Comment (cpsievert) [inline .github/internal/install-shinyverse/action.yaml:64]:
```suggestion
```

File: .github/internal/install-shinyverse/action.yaml [hunk 9]
```diff
@@ -40,54 +48,43 @@ runs:
...
```
Comment (cpsievert) [inline .github/internal/install-shinyverse/action.yaml:65]:
```suggestion
```

File: .github/internal/install-shinyverse/action.yaml [hunk 10]
```diff
@@ -20,7 +20,15 @@ inputs:
...
+    default: "3.x"
+    required: false
+  packagemanager-cran-url:
+    description: packagemanager url to be used for CRAN installations
+    default: "https://packagemanager.posit.co/cran/latest"
+    required: false
+  cran-url:
+    description: CRAN url to be used for CRAN installations
+    default: "https://cran.rstudio.com"
     required: false
```
Comment (cpsievert) [inline .github/internal/install-shinyverse/action.yaml:32]:
```suggestion
```

File: .github/internal/install-shinyverse/action.yaml [hunk 11]
```diff
@@ -40,54 +48,28 @@ runs:
...
-        message("Final packages:\n", gsub(",", ",\n", pkgs_to_install))
-        cat("to-install=", pkgs_to_install, "\n", file = Sys.getenv("GITHUB_OUTPUT"), sep = "", append = TRUE)
-
-    - name: Install R, shinycoreci, and shinyverse
+    # We need to have the first repo to be `shinyverse` repo, then RSPM, then cran-url
+    # Recreate logic from https://github.com/r-lib/actions/blob/413b6a4ba22b7bcaa726df1c8f88dcd574fbc8c4/setup-r/src/installer.ts#L598-L670
+    # * Do not set RSPM input (use-public-rspm: false) to make `CRAN` repo the first repo
+    # * Set CRAN env so that `packagemanger` repo is first
+    # * Add backup of `cran-url` in `extra-repositories`
+    # * Do not set RSPM envvar pointing to `packagemanager-cran-url` as it will make `RSPM` repo the first repo
```
Comment (cpsievert) [inline .github/internal/install-shinyverse/action.yaml:56]:
```suggestion
```

File: .github/internal/install-shinyverse/action.yaml [hunk 12]
```diff
@@ -40,54 +48,28 @@ runs:
...
+    - name: Install R and shinycoreci
       uses: rstudio/shiny-workflows/setup-r-package@v1
+      env:
+        CRAN: ${{ inputs.packagemanager-cran-url }}
```
Comment (cpsievert) [inline .github/internal/install-shinyverse/action.yaml:60]:
```suggestion
```

File: .github/internal/install-shinyverse/action.yaml [hunk 13]
```diff
@@ -40,54 +48,28 @@ runs:
...
-          ${{ steps.pkgs.outputs.to-install }}
           ${{ inputs.extra-packages }}
+        use-public-rspm: false
+        extra-repositories: |
+          ${{ inputs.cran-url }}
+
+    - name: Show repos
+      shell: Rscript {0}
+      run: |
+        print(getOption("repos"))
```
Comment (cpsievert) [inline .github/internal/install-shinyverse/action.yaml:75]:
```suggestion
```

File: .github/workflows/apps-config.yml [hunk 14]
```diff
@@ -16,9 +16,18 @@ on:
...
         value: "ubuntu-20.04"
+      ubuntu-packagemanager-cran-url:
+        description: ubuntu packagemanager url to be used for CRAN installations
+        value: "https://packagemanager.posit.co/cran/__linux__/focal/latest"
+      packagemanager-cran-url:
+        description: packagemanager url to be used for CRAN installations
+        value: "https://packagemanager.posit.co/cran/latest"
+      shinyverse-cran-url:
+        description: CRAN url to be used for shinyverse installations via r-universe
+        value: "https://posit-dev-shinycoreci.r-universe.dev"
```
Comment (cpsievert) [inline .github/workflows/apps-config.yml:27]:
```suggestion
```

File: .github/workflows/apps-deploy.yml [hunk 15]
```diff
@@ -65,12 +65,14 @@ jobs:
...
-        timeout-minutes: 45
+        timeout-minutes: 30 # Should be < 10 mins
         uses: ./.github/internal/install-shinyverse
         with:
           r-version: ${{ matrix.config.r }}
           cache-version: deploy-2-${{ needs.config.outputs.cache-version }}
-          extra-packages: rstudio/rsconnect
+          # extra-packages: rstudio/rsconnect
+          packagemanager-cran-url: ${{ matrix.config.packagemanager-cran-url}}
+          shinyverse-cran-url: ${{ needs.config.outputs.shinyverse-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-deploy.yml:75]:
```suggestion
```

File: .github/workflows/apps-deploy.yml [hunk 16]
```diff
@@ -37,26 +37,26 @@ jobs:
             cores: 5
             retry: 3
             # extra_app_text: ", apps = c('000-all', '000-manual', '300-bs-themer')"
+            packagemanager-cran-url: ${{ needs.config.outputs.ubuntu-packagemanager-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-deploy.yml:40]:
```suggestion
```

File: .github/workflows/apps-deploy.yml [hunk 17]
```diff
@@ -37,26 +37,26 @@ jobs:
...
             r: "${{ needs.config.outputs.oldrel1 }}"
             type: "connect"
             account: "barret"
             server_name: "rsc.radixu.com"
             server_url: "https://rsc.radixu.com/__api__"
-            # rspm: "https://demo.rstudiopm.com/all/__linux__/focal/latest"
             cores: 1
             retry: 3
             # extra_app_text: ", apps = c('000-all', '000-manual', '300-bs-themer')"
+            packagemanager-cran-url: ${{ needs.config.outputs.ubuntu-packagemanager-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-deploy.yml:51]:
```suggestion
```

File: .github/workflows/apps-test-matrix.yml [hunk 18]
```diff
@@ -25,184 +25,157 @@ jobs:
...
     uses: ./.github/workflows/apps-test-os.yml
     with:
       r-version: ${{ needs.config.outputs.release }}
       os: ${{ needs.config.outputs.macos }}
       cache-version: ${{ needs.config.outputs.cache-version }}
-    secrets:
-      SLACK_CHANNEL_ID: ${{ secrets.SLACK_CHANNEL_ID }}
-      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      packagemanager-cran-url: ${{ needs.config.outputs.packagemanager-cran-url }}
+      shinyverse-cran-url: ${{ needs.config.outputs.shinyverse-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-test-matrix.yml:35]:
```suggestion
```

File: .github/workflows/apps-test-matrix.yml [hunk 19]
```diff
@@ -25,184 +25,157 @@ jobs:
...
     uses: ./.github/workflows/apps-test-os.yml
     with:
       r-version: ${{ needs.config.outputs.oldrel1 }}
       os: ${{ needs.config.outputs.macos }}
       cache-version: ${{ needs.config.outputs.cache-version }}
-    secrets:
-      SLACK_CHANNEL_ID: ${{ secrets.SLACK_CHANNEL_ID }}
-      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      packagemanager-cran-url: ${{ needs.config.outputs.packagemanager-cran-url }}
+      shinyverse-cran-url: ${{ needs.config.outputs.shinyverse-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-test-matrix.yml:44]:
```suggestion
```

File: .github/workflows/apps-test-matrix.yml [hunk 20]
```diff
@@ -25,184 +25,157 @@ jobs:
...
     uses: ./.github/workflows/apps-test-os.yml
     with:
       r-version: ${{ needs.config.outputs.oldrel2 }}
       os: ${{ needs.config.outputs.macos }}
       cache-version: ${{ needs.config.outputs.cache-version }}
-    secrets:
-      SLACK_CHANNEL_ID: ${{ secrets.SLACK_CHANNEL_ID }}
-      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      packagemanager-cran-url: ${{ needs.config.outputs.packagemanager-cran-url }}
+      shinyverse-cran-url: ${{ needs.config.outputs.shinyverse-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-test-matrix.yml:53]:
```suggestion
```

File: .github/workflows/apps-test-matrix.yml [hunk 21]
```diff
@@ -25,184 +25,157 @@ jobs:
...
     uses: ./.github/workflows/apps-test-os.yml
     with:
       r-version: ${{ needs.config.outputs.oldrel3 }}
       os: ${{ needs.config.outputs.macos }}
       cache-version: ${{ needs.config.outputs.cache-version }}
-    secrets:
-      SLACK_CHANNEL_ID: ${{ secrets.SLACK_CHANNEL_ID }}
-      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      packagemanager-cran-url: ${{ needs.config.outputs.packagemanager-cran-url }}
+      shinyverse-cran-url: ${{ needs.config.outputs.shinyverse-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-test-matrix.yml:62]:
```suggestion
```

File: .github/workflows/apps-test-matrix.yml [hunk 22]
```diff
@@ -25,184 +25,157 @@ jobs:
...
     uses: ./.github/workflows/apps-test-os.yml
     with:
       r-version: ${{ needs.config.outputs.oldrel4 }}
       os: ${{ needs.config.outputs.macos }}
       cache-version: ${{ needs.config.outputs.cache-version }}
-    secrets:
-      SLACK_CHANNEL_ID: ${{ secrets.SLACK_CHANNEL_ID }}
-      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      packagemanager-cran-url: ${{ needs.config.outputs.packagemanager-cran-url }}
+      shinyverse-cran-url: ${{ needs.config.outputs.shinyverse-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-test-matrix.yml:71]:
```suggestion
```

File: .github/workflows/apps-test-matrix.yml [hunk 23]
```diff
@@ -25,184 +25,157 @@ jobs:
...
     uses: ./.github/workflows/apps-test-os.yml
     with:
       r-version: ${{ needs.config.outputs.release }}
       os: ${{ needs.config.outputs.ubuntu }}
       cache-version: ${{ needs.config.outputs.cache-version }}
-    secrets:
-      SLACK_CHANNEL_ID: ${{ secrets.SLACK_CHANNEL_ID }}
-      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      packagemanager-cran-url: ${{ needs.config.outputs.ubuntu-packagemanager-cran-url }}
+      shinyverse-cran-url: ${{ needs.config.outputs.shinyverse-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-test-matrix.yml:81]:
```suggestion
```

File: .github/workflows/apps-test-matrix.yml [hunk 24]
```diff
@@ -25,184 +25,157 @@ jobs:
...
     uses: ./.github/workflows/apps-test-os.yml
     with:
       r-version: ${{ needs.config.outputs.oldrel1 }}
       os: ${{ needs.config.outputs.ubuntu }}
       cache-version: ${{ needs.config.outputs.cache-version }}
-    secrets:
-      SLACK_CHANNEL_ID: ${{ secrets.SLACK_CHANNEL_ID }}
-      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      packagemanager-cran-url: ${{ needs.config.outputs.ubuntu-packagemanager-cran-url }}
+      shinyverse-cran-url: ${{ needs.config.outputs.shinyverse-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-test-matrix.yml:90]:
```suggestion
```

File: .github/workflows/apps-test-matrix.yml [hunk 25]
```diff
@@ -25,184 +25,157 @@ jobs:
...
     uses: ./.github/workflows/apps-test-os.yml
     with:
       r-version: ${{ needs.config.outputs.oldrel3 }}
       os: ${{ needs.config.outputs.ubuntu }}
       cache-version: ${{ needs.config.outputs.cache-version }}
-    secrets:
-      SLACK_CHANNEL_ID: ${{ secrets.SLACK_CHANNEL_ID }}
-      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      packagemanager-cran-url: ${{ needs.config.outputs.ubuntu-packagemanager-cran-url }}
+      shinyverse-cran-url: ${{ needs.config.outputs.shinyverse-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-test-matrix.yml:108]:
```suggestion
```

File: .github/workflows/apps-test-matrix.yml [hunk 26]
```diff
@@ -25,184 +25,157 @@ jobs:
...
     uses: ./.github/workflows/apps-test-os.yml
     with:
       r-version: ${{ needs.config.outputs.oldrel4 }}
       os: ${{ needs.config.outputs.ubuntu }}
       cache-version: ${{ needs.config.outputs.cache-version }}
-    secrets:
-      SLACK_CHANNEL_ID: ${{ secrets.SLACK_CHANNEL_ID }}
-      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      packagemanager-cran-url: ${{ needs.config.outputs.ubuntu-packagemanager-cran-url }}
+      shinyverse-cran-url: ${{ needs.config.outputs.shinyverse-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-test-matrix.yml:117]:
```suggestion
```

File: .github/workflows/apps-test-matrix.yml [hunk 27]
```diff
@@ -25,184 +25,157 @@ jobs:
...
     uses: ./.github/workflows/apps-test-os.yml
     with:
       r-version: ${{ needs.config.outputs.release }}
       os: ${{ needs.config.outputs.windows }}
       cache-version: ${{ needs.config.outputs.cache-version }}
-    secrets:
-      SLACK_CHANNEL_ID: ${{ secrets.SLACK_CHANNEL_ID }}
-      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      packagemanager-cran-url: ${{ needs.config.outputs.packagemanager-cran-url }}
+      shinyverse-cran-url: ${{ needs.config.outputs.shinyverse-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-test-matrix.yml:127]:
```suggestion
```

File: .github/workflows/apps-test-matrix.yml [hunk 28]
```diff
@@ -25,184 +25,157 @@ jobs:
...
     uses: ./.github/workflows/apps-test-os.yml
     with:
       r-version: ${{ needs.config.outputs.oldrel1 }}
       os: ${{ needs.config.outputs.windows }}
       cache-version: ${{ needs.config.outputs.cache-version }}
-    secrets:
-      SLACK_CHANNEL_ID: ${{ secrets.SLACK_CHANNEL_ID }}
-      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      packagemanager-cran-url: ${{ needs.config.outputs.packagemanager-cran-url }}
+      shinyverse-cran-url: ${{ needs.config.outputs.shinyverse-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-test-matrix.yml:136]:
```suggestion
```

File: .github/workflows/apps-test-matrix.yml [hunk 29]
```diff
@@ -25,184 +25,157 @@ jobs:
...
     uses: ./.github/workflows/apps-test-os.yml
     with:
       r-version: ${{ needs.config.outputs.oldrel2 }}
       os: ${{ needs.config.outputs.windows }}
       cache-version: ${{ needs.config.outputs.cache-version }}
-    secrets:
-      SLACK_CHANNEL_ID: ${{ secrets.SLACK_CHANNEL_ID }}
-      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      packagemanager-cran-url: ${{ needs.config.outputs.packagemanager-cran-url }}
+      shinyverse-cran-url: ${{ needs.config.outputs.shinyverse-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-test-matrix.yml:145]:
```suggestion
```

File: .github/workflows/apps-test-matrix.yml [hunk 30]
```diff
@@ -25,184 +25,157 @@ jobs:
...
     uses: ./.github/workflows/apps-test-os.yml
     with:
       r-version: ${{ needs.config.outputs.oldrel3 }}
       os: ${{ needs.config.outputs.windows }}
       cache-version: ${{ needs.config.outputs.cache-version }}
-    secrets:
-      SLACK_CHANNEL_ID: ${{ secrets.SLACK_CHANNEL_ID }}
-      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      packagemanager-cran-url: ${{ needs.config.outputs.packagemanager-cran-url }}
+      shinyverse-cran-url: ${{ needs.config.outputs.shinyverse-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-test-matrix.yml:154]:
```suggestion
```

File: .github/workflows/apps-test-matrix.yml [hunk 31]
```diff
@@ -25,184 +25,157 @@ jobs:
...
     uses: ./.github/workflows/apps-test-os.yml
     with:
       r-version: ${{ needs.config.outputs.oldrel4 }}
       os: ${{ needs.config.outputs.windows }}
       cache-version: ${{ needs.config.outputs.cache-version }}
-    secrets:
-      SLACK_CHANNEL_ID: ${{ secrets.SLACK_CHANNEL_ID }}
-      SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+      packagemanager-cran-url: ${{ needs.config.outputs.packagemanager-cran-url }}
+      shinyverse-cran-url: ${{ needs.config.outputs.shinyverse-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-test-matrix.yml:163]:
```suggestion
```

File: .github/workflows/apps-test-os.yml [hunk 32]
```diff
@@ -27,17 +27,17 @@ on:
...
       pandoc-version:
         type: string
-        default: "2.7.3"
+        default: "3.x"
+        required: false
+      packagemanager-cran-url:
+        type: string
+        description: packagemanager url to be used for CRAN installations
+        default: "https://packagemanager.posit.co/cran/latest"
         required: false
```
Comment (cpsievert) [inline .github/workflows/apps-test-os.yml:36]:
```suggestion
```

File: .github/workflows/apps-test-os.yml [hunk 33]
```diff
@@ -126,39 +126,18 @@ jobs:
...
-        timeout-minutes: 100 # windows may take a while
+        timeout-minutes: 30 # Should be < 10 mins
         uses: ./.github/internal/install-shinyverse
         with:
           r-version: ${{ inputs.r-version }}
           cache-version: ${{ inputs.cache-version }}
           http-user-agent: ${{ inputs.http-user-agent }}
           pandoc-version: ${{ inputs.pandoc-version }}
           extra-packages: ${{ inputs.extra-packages }}
+          packagemanager-cran-url: ${{ inputs.packagemanager-cran-url }}
```
Comment (cpsievert) [inline .github/workflows/apps-test-os.yml:138]:
```suggestion
```

File: inst/apps/000-pkg-versions/app.R [hunk 34]
```diff
@@ -0,0 +1,115 @@
...
+    )
+  }
+
+  dt <- bind_rows(lapply(pkgs, get_pkg_info))
+
+  print(dt, n = Inf, width = 1000)
+
+  # ----------------------------
+
+  bad_dt <- dt %>% filter(!is.na(remoteRepos)) %>% filter(remoteRepos != universe_url)
```
Comment (cpsievert) [inline inst/apps/000-pkg-versions/app.R:80]:
```suggestion
  # Packages inside `dt` are in the shinyverse so if there remote location is 
  # not the universe url, then they're not properly installed.
  bad_dt <- dt %>% filter(!is.na(remoteRepos)) %>% filter(remoteRepos != universe_url)
```

**Comment by schloerke** [comment]
🤞

**Pull request merged**